cpp/include/cuml/fil/multireduction.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+// data [T]ype, reduction [R]adix
+template <int R, typename T>
+__device__ T multireduction(T* per_thread, const int set_size, int n_sets) {
```

Comment:
Consider renaming `n_sets` to `n_values` and `set_size` to `n_groups` (if I've guessed the meaning of each parameter correctly).